core/src/Lender.sol

```diff
@@ -50,23 +50,32 @@ contract Lender is Ledger {
 
     constructor(address reserve) Ledger(reserve) {}
 
-    function initialize(IRateModel rateModel_, uint8 reserveFactor_) external {
+    function initialize() external {
         require(borrowIndex == 0);
         borrowIndex = uint72(ONE);
         lastAccrualTime = uint32(block.timestamp);
 
         initialDomainSeparator = _computeDomainSeparator();
         initialChainId = block.chainid;
+    }
 
+    /// @notice Sets the `rateModel` and `reserveFactor`. Only the `FACTORY` can call this.
+    function setRateModelAndReserveFactor(IRateModel rateModel_, uint8 reserveFactor_) external {
+        require(msg.sender == address(FACTORY));
         rateModel = rateModel_;
-        require(MIN_RESERVE_FACTOR <= reserveFactor_ && reserveFactor_ <= MAX_RESERVE_FACTOR);
         reserveFactor = reserveFactor_;
```

Comment:
Should we emit an event after setting the rate model and reserve factor?

Reply:
Yes. Good catch; I'll make sure to add this when I add the timelock (1 or 2 PR's down the line)